parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java

```diff
@@ -1303,4 +1352,111 @@ public void resumeIfPaused() {
         }
     }
 
+    /**
+     * An inbound message to the controller.
+     * <p>
+     * Currently, an Either type class, representing either newly polled records to ingest, or a work result.
+     */
+    @Value
+    @RequiredArgsConstructor(access = PRIVATE)
+    private static class ControllerEventMessage<K, V> {
+
+        WorkContainer<K, V> workContainer;
+
+        EpochAndRecordsMap<K, V> consumerRecords;
+
+        private boolean isWorkResult() {
+            return workContainer != null;
+        }
+
+        private boolean isNewConsumerRecords() {
+            return !isWorkResult();
+        }
+
+
+//        CompositeMeterRegistry compositeRegistry = new CompositeMeterRegistry();
+
+//        compositeRegistry.add(oneSimpleMeter);
+//
+//        var registry = compositeRegistry;
+//
+//
+//        successCounter.increment(2.0);
+//
+//        assertTrue(successCounter.count() == 2);
+//
+//        successCounter.increment(-1);
+//
+//        assertTrue(successCounter.count() == 1);
+//
+//
+//        io.micrometer.core.instrument.Timer timer = registry.timer("app.event");
+//        timer.record(() -> {
+//            try {
+//                TimeUnit.MILLISECONDS.sleep(15);
+//            } catch (InterruptedException ignored) {
+//            }
+//        });
+//
+//        timer.record(30, TimeUnit.MILLISECONDS);
+//
+//
+//        SimpleMeterRegistry registry = new SimpleMeterRegistry();
+//        List<String> list = new ArrayList<>(4);
+//
+//        Gauge gauge = Gauge
+//                .builder("cache.size", list, List::size)
+//                .register(registry);
+//
+//
+//        list.add("1");
+//
+//
+//        //
+//        DistributionSummary distributionSummary = DistributionSummary
+//                .builder("request.size")
+//                .baseUnit("bytes")
+//                .register(registry);
+//
+//        distributionSummary.record(3);
+//        distributionSummary.record(4);
+//        distributionSummary.record(5);
+//
+//
+//        Timer timer = new Timer
+//                .("test.timer")
+//                .publishPercentiles(0.3, 0.5, 0.95)
+//                .publishPercentileHistogram()
+//                .register(registry);
+
+
+       // new LogbackMetrics().bindTo(metricsRegistry);
+
+        private static <K, V> ControllerEventMessage<K, V> of(EpochAndRecordsMap<K, V> polledRecords) {
+            return new ControllerEventMessage<>(null, polledRecords);
+        }
+
+        public static <K, V> ControllerEventMessage<K, V> of(WorkContainer<K, V> work) {
+            return new ControllerEventMessage<K, V>(work, null);
+        }
+    }
+
+    /**
+     * todo docs
+     */
+    public PCMetrics calculateMetricsWithIncompletes() {
```

Comment:
as above, what should our health end point look like, and does it makes sense to have direct access like this?